holidayextras/plugin-jsonapi
# Web 9248

#### What does this PR do? (please provide any background)
[WEB-9248](https://hxshortbreaks.atlassian.net/browse/WEB-9248)
Updating the code to our latest coding guidelines.

#### What tests does this PR have?
pluginJsonapi
    #errorHandler
      ✓ should call the request log with an error
    #alsoMakeItSo
      ✓ should return early when isBoom is true
      ✓ should throw an error when the request.route.settings.bind.resourceName is not set
      ✓ should return early when no resources are found
utilities
    #collectIncludes
      ✓ should collect the include and add it to the stateObject
      ✓ should return foo and bar in an array on the stateObject
      ✓ should return early as no include found
    #getResources
      ✓ should return early as there are no ids on the resource
      ✓ should build a href for the resource
    #getSubResources
      ✓ should add a sub resource
      ✓ should not attach any sub resources as nothing is included
    #resolveLinkedData
      ✓ should return early as no sub reources are provided
      ✓ should attach the sub reources as linked
    #_buildHref
      ✓ should return an URL the type and id
      ✓ should return an URL the type and the ids as comma separated list
      ✓ should return an URL the type and the filter added as a query string
      ✓ should return an URL the type, ids and the filter added as a query string
    #_addSubResourceRequest
      ✓ should add the link to the subResources
      ✓ should add a new id to the already existing subResource
      ✓ should add the link to the subResources (new filter)
      ✓ should add a new filter id to the alredy existed subResource (new filter)
    #_getResourcesLinks
      ✓ should build the href for the resource
      ✓ should build the href for the resource and remove the filter

#### How can this be tested?
In **package.json** the Works change the version of plugin-jsonapi to this branch:
```
"plugin-jsonapi": "git+ssh://git@github.com:holidayextras/plugin-jsonapi.git#WEB-9248",
```
Then install the node packages.
Once installed, you should be able to hit any endpoint on the Works and results back, compare against the Works production.

#### Any tech debt?
None

#### Screenshots / Screencast
None

#### What gif best describes how you feel about this work?
![]()


---

**Reviewer**

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

---
